comp/core/config/config.go

```diff
@@ -69,10 +78,18 @@ func NewServerlessConfig(path string) (Component, error) {
 	return newConfig(d)
 }
 
-func newConfig(deps dependencies) (Component, error) {
+func newConfigProvide(deps dependencies) (provides, error) {
```

Comment:
We're in the `config` package, no need to repeat `config` in methods name.
```suggestion
func newComponent(deps dependencies) (provides, error) {
```

Reply:
Good point ! Done in c87e3fdcf2ba9ab7d45960217df9bd31db89429d.